integrated_channels/integrated_channel/models.py

```diff
@@ -60,5 +78,133 @@ class EnterpriseCustomerPluginConfiguration(TimeStampedModel):
     )
     active = models.BooleanField(blank=False, null=False)
 
+    GRADE_PASSING = 'Pass'
+    GRADE_FAILING = 'Fail'
+    GRADE_UNKNOWN = 'In Progress'
+
     class Meta:
         abstract = True
+
+    @staticmethod
+    def channel_code():
+        """
+        Returns an capitalized identifier for this channel class, unique among subclasses.
+        """
+        raise NotImplementedError(_('Implemented in concrete subclass.'))
+
+    def transmit_learner_data(self):
+        """
+        Collect and transmit learner data for the ``EnterpriseCustomer``.
+        """
+        transmitter = self.get_learner_data_transmitter()
+        for learner_data in self.collect_learner_data():
+            transmitter.transmit(learner_data)
+
+    def collect_learner_data(self):
+        """
+        Collect learner data for the ``EnterpriseCustomer`` where data sharing consent is granted.
+
+        Yields a learner data object for each enrollment, containing:
+
+        * ``enterprise_enrollment``: ``EnterpriseCourseEnrollment`` object.
+        * ``certificate``: ``GeneratedCertificate`` for the user+course; None if not found.
+          "Course completion" occurs when course certificates are issued.
+        """
+
```

Comment:
I'm not sure what is simplest.. the consent logic is getting pretty tangled.

We could certainly check here.  We do say pretty clearly on [`EnterpriseCustomer.enable_data_sharing_consent`](https://github.com/edx/edx-enterprise/blob/864030005bb8c84b2b996399fbaca4c9adc02561/enterprise/models.py#L101-L107):
> This field is used to determine whether data sharing consent is enabled or disabled for users signing in using this enterprise customer. If disabled, consent will not be requested, and eligible data will not be shared.

And/or filter out the `EnterpriseCustomer`s without data sharing enabled [in the management command filter](https://github.com/edx/edx-enterprise/pull/69/files/835e06a373651c57c2fc0165f4e3287ac20efaae#diff-11c9715fd5316220a126561ca420c9dcR100).

And/or, we could [change this check below](https://github.com/edx/edx-enterprise/pull/69/files/835e06a373651c57c2fc0165f4e3287ac20efaae#diff-4386ccd909983c900b04b5a992e66fddR134) to something more specific for each enrollment.  I started off checking `enrollment.consent_granted`, but then changed it to `enrollment.consent_available()`, which took the [`UserDataSharingConsentAudit`](https://github.com/edx/edx-enterprise/blob/864030005bb8c84b2b996399fbaca4c9adc02561/enterprise/models.py#L574) records into account as well.